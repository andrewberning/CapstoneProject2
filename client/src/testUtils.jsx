import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
