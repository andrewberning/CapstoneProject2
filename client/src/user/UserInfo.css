.UserInfo {
  max-width: 600px;
  margin: 40px auto;
  padding: 20px;
  background-color: #f8f9fa;
  border-radius: 8px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
}

.UserInfo h1 {
  text-align: center;
  color: #333;
  margin-bottom: 20px;
  font-size: 24px;
}

.user-info-container {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.user-info-container p {
  margin: 0;
  padding: 10px;
  background-color: #ffffff;
  border: 1px solid #e0e0e0;
  border-radius: 4px;
  font-size: 16px;
  color: #555;
}

.user-info-container p:nth-child(odd) {
  background-color: #f1f1f1;
}

@media (max-width: 480px) {
  .UserInfo {
    padding: 15px;
  }

  .UserInfo h1 {
    font-size: 20px;
  }

  .user-info-container p {
    font-size: 14px;
  }
}
