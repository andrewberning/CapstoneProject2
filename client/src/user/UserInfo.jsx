import { useContext } from "react";
import UserContext from "../auth/UserContext";
import "./UserInfo.css";


export default function UserInfo() {
  const {currentUser} = useContext(UserContext);

  return (
    <div className="UserInfo">
      <h1>User Information</h1>
      <div className="user-info-contaienr">
        <p>First name: {currentUser.firstName}</p>
        <p>Last name: {currentUser.lastName}</p>
        <p>Username: {currentUser.username}</p>
        <p>Email: {currentUser.email}</p>
      </div>
    </div>

  )
}