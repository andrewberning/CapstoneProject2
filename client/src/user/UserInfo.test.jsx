import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserInfo from "./UserInfo";
import { UserProvider } from "../testUtils";

describe('UserInfo component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <UserInfo />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <UserInfo />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
})