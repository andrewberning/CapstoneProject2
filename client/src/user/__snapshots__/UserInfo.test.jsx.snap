// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserInfo component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="UserInfo"
  >
    <h1>
      User Information
    </h1>
    <div
      class="user-info-contaienr"
    >
      <p>
        First name: 
      </p>
      <p>
        Last name: 
      </p>
      <p>
        Username: testuser
      </p>
      <p>
        Email: testuser123@test.net
      </p>
    </div>
  </div>
</DocumentFragment>
`;
